includes/reader-revenue/my-account/class-woocommerce-my-account.php

```diff
@@ -131,16 +150,67 @@ public static function handle_password_reset_request() {
 		}
 	}
 
+	/**
+	 * Handle magic link request.
+	 */
+	public static function handle_magic_link_request() {
+		if ( ! is_user_logged_in() ) {
+			return;
+		}
+		$nonce = filter_input( INPUT_GET, self::SEND_MAGIC_LINK_PARAM, FILTER_SANITIZE_STRING );
+
+		if ( $nonce ) {
+			$is_error = false;
+			if ( wp_verify_nonce( $nonce, self::SEND_MAGIC_LINK_PARAM ) ) {
+				$result  = Reader_Activation::send_verification_email( wp_get_current_user() );
+				$message = __( 'Please check your email inbox for a link to verify your account.', 'newspack' );
+				if ( is_wp_error( $result ) ) {
+					Logger::log( 'Error sending verification email: ' . $result->get_error_message() );
+					$message  = __( 'Something went wrong.', 'newspack' );
+					$is_error = true;
+				}
+			} else {
+				$message  = __( 'Something went wrong.', 'newspack' );
+				$is_error = true;
+			}
+			wp_safe_redirect(
+				add_query_arg(
+					[
+						'message'  => $message,
+						'is_error' => $is_error,
+					],
+					remove_query_arg( self::SEND_MAGIC_LINK_PARAM )
+				)
+			);
+			exit;
+		}
+	}
+
+	/**
+	 * Check if the user is logged in and verified.
+	 */
+	public static function is_reader_verified() {
+		// Don't lock access if Reader Activation features aren't enabled.
+		if ( ! Reader_Activation::is_enabled() ) {
+			return true;
+		}
+
+		return \is_user_logged_in() && Reader_Activation::is_reader_verified( \wp_get_current_user() );
+	}
+
 	/**
 	 * Redirect to "Account details" if accessing "My Account" directly.
 	 */
 	public static function redirect_to_account_details() {
 		if ( Donations::is_platform_stripe() && function_exists( 'wc_get_page_permalink' ) ) {
 			global $wp;
-			$current_url               = home_url( $wp->request );
-			$my_account_page_permalink = wc_get_page_permalink( 'myaccount' );
-			if ( trailingslashit( $current_url ) === trailingslashit( $my_account_page_permalink ) ) {
-				wp_safe_redirect( '/my-account/edit-account/' );
+			$current_url                  = home_url( $wp->request );
+			$my_account_page_permalink    = wc_get_page_permalink( 'myaccount' );
+			$my_account_details_permalink = wc_get_account_endpoint_url( 'edit-account' );
+			$is_my_account_root           = trailingslashit( $current_url ) === trailingslashit( $my_account_page_permalink );
+			$is_my_account_details        = trailingslashit( $current_url ) === trailingslashit( $my_account_details_permalink );
+			if ( $is_my_account_root || ( ! $is_my_account_details && ! self::is_reader_verified() ) ) {
```

Comment:
```suggestion
			$is_my_account_root           = trailingslashit( $current_url ) === trailingslashit( $my_account_page_permalink );
			if ( $is_my_account_root && ! self::is_reader_verified() ) {
```

The current check is redirecting every anonymous page view to `/my-account/edit-account`

Reply:
Unfortunately the suggestion broke the behavior to block access to non-root account pages if unverified (so it was still possible to visit locked pages by direct URL input). 789d6d3 should fix this and also avoid redirecting when not on an account page.